impl/src/main/java/org/pentaho/platform/genericfile/providers/repository/RepositoryFileProvider.java

```diff
@@ -492,6 +493,36 @@ public List<IGenericFile> getDeletedFiles() {
       .collect( Collectors.toList() );
   }
 
+  @Override
+  public void deleteFilePermanently( @NonNull GenericFilePath path ) throws OperationFailedException {
+    String fileId = getFileId( path );
+
+    try {
+      fileService.doDeleteFilesPermanent( fileId );
+    } catch ( Exception e ) {
+      throw new OperationFailedException( e );
+    }
+  }
+
+  protected String getFileId( @NonNull GenericFilePath path ) throws InvalidPathException {
+    List<String> segments = path.getSegments();
+
+    for ( int i = 0; i < segments.size(); i++ ) {
+      if ( FOLDER_NAME_TRASH.equals( segments.get( i ) ) && i + 1 < segments.size() ) {
+        String segment = segments.get( i + 1 );
+        int colonIndex = segment.indexOf( ':' );
+
+        if ( colonIndex != -1 ) {
+          return segment.substring( colonIndex + 1 );
+        } else {
+          return segment;
```

Comment:
I don't think we should be lenient here. As far as we know, the above case's format is the correct one.